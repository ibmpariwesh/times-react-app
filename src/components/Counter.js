

export default function Counter(props){//stateless
    return (
        <span>{props.count +1}</span>
    )
}