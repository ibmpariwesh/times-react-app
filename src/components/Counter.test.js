import { render, screen } from '@testing-library/react';
import Counter from "./Counter";

test('scenario 1', ()=>{
    render(<Counter  count='5'/>);
  const element = screen.getByText("5");
    console.log('testing....');
    expect(element.innerHTML).toBe("5");
    expect(element.tagName).toEqual('SPAN');
})

xtest('testing using snapshot', ()=>{
    const element  = render(<Counter  count='5'/>).asFragment;
    expect(element()).toMatchSnapshot();
})

xtest('testing using snapshot with count=10', ()=>{
    const element  = render(<Counter  count='10'/>).asFragment;
    expect(element()).toMatchSnapshot();
})