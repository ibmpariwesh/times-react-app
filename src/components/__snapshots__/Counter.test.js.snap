// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing using snapshot 1`] = `
<DocumentFragment>
  <span>
    5
  </span>
</DocumentFragment>
`;

exports[`testing using snapshot with count=10 1`] = `
<DocumentFragment>
  <span>
    10
  </span>
</DocumentFragment>
`;
